### PR TITLE
fix failed tests in numpy2.0

### PR DESCRIPTION
**Pull request by risemeup1**
### PR Category

Others

### PR Types

Others
### Description

将单测里的requriements.txt和python/requirements.txt合二为一，并修复以下单测在numpy2.x下的错误

1. test_solve_op:在numpy2.x中np.linalg.solve函数似乎不支持批量处理，np.linalg.solve期望输入矩阵X的形状应该是（m, m），常数矩阵Y的形状应该是（m, n)，当输入是3D的，比如（20, 6, 6），它也会认为前两个维度定义20,6的矩阵是方阵，所以导致报错了，在此处通过for循环对每一对系数矩阵和常数向量都调用np.linalg.solve
image

2. test_bitwise_shift_op: int8的取值范围是-128到127。如果255赋值给int8类型的变量，这显然超出了它的取值范围，因此引发了OverflowError。所以改成int16
image

3. gymnasium和wandb升级版本到支持numpy2.0
4. test_unique:numpy.unique函数在2.x版本中的改动，让inverse返回的形状与原始输入数组相同，在1.x下返回的却是flatten之后的结果，例如：
`x=np.random.randint(0, 100, (4, 7, 10)) ;
unique, indices, inverse, counts = np.unique(x,return_index=True,return_inverse=True,return_counts=True,axis=None,) 
在numpy1.x下inverse.shape是(280,)
在numpy2.x下是(4,7,10)`

Pcard-67164

### CODE REVIEW

**Review by SigureMo — DISMISSED** [review]
LGTMeow

File: test/deprecated/legacy_test/test_learning_rate_scheduler.py [hunk 1]
```diff
@@ -570,6 +570,7 @@ def run_scalar_lr(self, place, lr, start_lr, end_lr):
             self.assertAlmostEqual(
                 expected_lr,
                 lr_val[0],
+                places=6,
```
Comment (chen2016013) [inline test/deprecated/legacy_test/test_learning_rate_scheduler.py:574]: numpy2.0会带来精度变化

**Review by chen2016013 — DISMISSED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]
